backend/app/routes.py

```diff
@@ -101,9 +103,30 @@ def get_links_between_two_nodes(from_node_id, to_node_id):
         abort(400, description="Source node can not be the same as target node.")
         return
 
-    connection = getConnection()
+    
+    links = get_link_geom(from_node_id, to_node_id)
 
-    with connection:
+
+    shortest_link_data = {
+        "source": from_node_id, 
+        "target": to_node_id,
+        "links": links,
+        # the following three fields are for compatibility and should eventually be removed
+        "path_name": "",
+        "link_dirs": [ link['link_dir'] for link in links ],
+        "geometry": {
+            "type": "MultiLineString",
+            "coordinates": [ link['geometry']['coordinates'] for link in links ]
+        }
+    }
+    return jsonify(shortest_link_data)
+
+
+#Function that returns a json with geometries of links between two nodes
+@app.route('/get-link-geom/<from_node_id>/<to_node_id>', methods=['GET'])
+def get_link_geom(from_node_id, to_node_id):

```

Comment:
I wonder if `get_links` would be a clearer name for this? It's returning a set of links, not a geom.